### PR TITLE
Bump `react-native`  to version `0.71.8`

### PR DESCRIPTION
**Pull request by fluiddot**
Bumps the React Native version to `0.71.8`. This also implied updating the module substitution block to reflect the new [process for fetching the React Native and Hermes modules](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0508-out-of-npm-artifacts.md#implemented-solution).

## How to test
1. Run the command `npm install`.
2. Run the command `./gradlew publishToMavenLocal -exclude-task prepareToPublishToS3`.
3. Observe that all libraries are built and published successfully in the local folder `~/.m2/repository/org/wordpress-mobile/react-native-libraries`.

### CODE REVIEW

File: build.gradle.kts
Comment (fluiddot) [inline build.gradle.kts]: This follows the process performed by the React Native Gradle plugin ([reference](https://github.com/facebook/react-native/blob/ea4724b37c9f78bd33daab547d6cc4f8b7f7dd81/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/DependencyUtils.kt#L45-L74), [docs](https://reactnative.dev/docs/react-native-gradle-plugin#what-is-the-plugin-doing-under-the-hood)). Additionally, as shared in [this proposal](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0508-out-of-npm-artifacts.md#implemented-solution), now the React Native and Hermes modules are published to Maven, similar to what we do in [“react-native-mirror-publisher” repository](https://github.com/wordpress-mobile/react-native-mirror-publisher).

File: build.gradle.kts
Comment (oguzkocer) [inline build.gradle.kts]: Since we are the ones setting the `reactNativeMinorVersion` in `package.json`, we probably don't need to do this check, right? Assuming that we are not going to roll back to a previous version, the minor version should always be `>= 71`.

File: build.gradle.kts
Comment (oguzkocer) [inline build.gradle.kts]: I think due to this `react-native` to `react-android` change, we need to stop using our own mirror from [react-native-mirror-publisher](https://github.com/wordpress-mobile/react-native-mirror-publisher). I think we can start using the artifacts published to `mavenCentral` and archive the `react-native-mirror-publisher` project.

If we keep using our own mirror, we might get duplicate class errors. Even if we don't, it wouldn't be great to have dependencies on both `react-android` & `react-native-mirror`.

The same goes for the `hermes-android`, of course.

@fluiddot What do you think?

P.S: If you all decide to go through with this ^ suggestion, please let me know if you need any help!

**Review by oguzkocer — APPROVED** [review]
@fluiddot If the changes in this PR work as expected, they look good to me. I did leave a suggestion about removing the `reactNativeMinorVersion` property and the `if` check it's used in as I don't think it's necessary.

I also left a suggestion about the `react-native-mirror` to `react-android` change.

Neither of these are blocking this PR, so feel free to merge it as is.

File: build.gradle.kts
Comment (fluiddot) [reply build.gradle.kts]: Good point. The React Native version is fully controlled by the `package.json` file, so we won't need this as we're updating RN to `0.71`. I'll remove it before merging 👍. I hope we don't need to roll back to a previous version, but if need so, we'll have this PR as guidance.

File: build.gradle.kts
Comment (fluiddot) [reply build.gradle.kts]: > I think due to this `react-native` to `react-android` change, we need to stop using our own mirror from [react-native-mirror-publisher](https://github.com/wordpress-mobile/react-native-mirror-publisher). I think we can start using the artifacts published to `mavenCentral` and archive the `react-native-mirror-publisher` project.

That's correct. From now on, we won't need the mirror repository to publish React Native and Hermes. Hence, we could archive `react-native-mirror-publisher` project. In any case, I'd wait until the RN upgrade is fully released to do so.

> If we keep using our own mirror, we might get duplicate class errors. Even if we don't, it wouldn't be great to have dependencies on both `react-android` & `react-native-mirror`.
> 
> The same goes for the `hermes-android`, of course.
> 
> @fluiddot What do you think?
> 
> P.S: If you all decide to go through with this ^ suggestion, please let me know if you need any help!

I completely agree. In fact, as part of the RN upgrade effort, I'm removing all references to the mirror in Gutenberg and WP-Android, in favor of using the ones published in `mavenCentral`.

File: build.gradle.kts
Comment (fluiddot) [reply build.gradle.kts]: I applied this suggestion in https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/23/commits/f76aaf60bb9830dcfec7f3b79eb15152dbf7c726.

**Review by dcalhoun — COMMENTED** [review]
Thank you for putting this together. I really appreciate the comments explaining code logic, as I am unfamiliar with this repository or its scripts. 

I wanted to share that I ran into the following error when following the testing instructions. I presume it is unrelated to these changes. WDYT?

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':react-native-clipboard'.
> SoftwareComponentInternal with name 'release' not found.
```

**Comment by dcalhoun** [comment]
> I wanted to share that I ran into the following error when following the testing instructions. I presume it is unrelated to these changes. WDYT?
> 
> ```
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> A problem occurred configuring project ':react-native-clipboard'.
> > SoftwareComponentInternal with name 'release' not found.
> ```

I realized this error occurred as I did not run `npm install` within the repository before executing the testing instructions. The build succeeded this time. 🎉 

I will open a separate PR updating the project README documenting setup steps.

**Comment by fluiddot** [comment]
> I release this error occurred as I did not run `npm install` within the repository before executing the testing instructions. The build succeeded this time. 🎉

Oh, sorry for that, I forgot to add that step. I'll update the PR's description.

> I will open a separate PR updating the project README documenting setup steps.

Yeah, good idea. I thought about that when working on this PR, as we don't have instructions for testing 👍 .

File: build.gradle.kts
Comment (oguzkocer) [reply build.gradle.kts]: That's great to hear!

File: build.gradle.kts
Comment (oguzkocer) [reply build.gradle.kts]: Thanks for the update @fluiddot - it all looks good to me 👍

**Comment by fluiddot** [comment]
> I realized this error occurred as I did not run `npm install` within the repository before executing the testing instructions. The build succeeded this time. 🎉

@dcalhoun based on [this comment](https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/23#issuecomment-1587649064), I assume you were able to test the PR, would you mind if I merge the PR without your explicit PR approval? Thanks!

**Review by dcalhoun — APPROVED** [review]
> @dcalhoun based on [this comment](https://github.com/wordpress-mobile/react-native-libraries-publisher/pull/23#issuecomment-1587649064), I assume you were able to test the PR, would you mind if I merge the PR without your explicit PR approval? Thanks!

Yes, I did test and you are welcome to merge. I only abstained from an approval due to the error and my own lack of expertise in this repository. In the future, feel free to always merge as you see fit if you have approval from someone else.

**Pull request merged**
